catalog/app/containers/AddToPackage/Provider.tsx

```diff
@@ -0,0 +1,34 @@
+import { basename } from 'path'
+
+import * as R from 'ramda'
+import * as React from 'react'
+
+import type * as Model from 'model'
+
+const Ctx = React.createContext<{
+  append: (file: Model.S3File) => void
+  entries: Record<string, Model.S3File>
+}>({
+  append: () => {},
+  entries: {},
+})
+
+interface ProviderProps {
+  children: React.ReactNode
+}
+
+export function Provider({ children }: ProviderProps) {
+  const [entries, setEntries] = React.useState({})
+  const append = React.useCallback((s3File) => {
+    setEntries(
+      R.mergeLeft({
+        [basename(s3File.key)]: s3File,
+      }),
+    )
+  }, [])
```

Comment:
we probably need a way to clear the state once we consume the data -- it could be either a combined method (e.g. `consume()`) which would return the entries and clear the state, or a dedicated method (e.g. `clear()`) which can be called manually after we consume the entries

Reply:
Makes sense, I'll introduce `clear` method called after Package's onSubmit

Reply:
I'll add in the next PR too